spotbugs-tests/src/test/java/edu/umd/cs/findbugs/SuppressionMatcherTest.java

```diff
@@ -0,0 +1,150 @@
+package edu.umd.cs.findbugs;
+
+import edu.umd.cs.findbugs.annotations.SuppressMatchType;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+import java.util.Collections;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.startsWith;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+class SuppressionMatcherTest {
+
+    private static final String PACKAGE_NAME = "com.example";
+    private static final String CLASS_NAME = PACKAGE_NAME + ".Test";
+    private static final ClassAnnotation CLASS_ANNOTATION = new ClassAnnotation(CLASS_NAME);
+
+    private SuppressionMatcher matcher;
+    private BugReporter bugReporter;
+    private ArgumentCaptor<BugInstance> bugCaptor;
+
+    @BeforeEach
+    void init() {
+        matcher = new SuppressionMatcher();
+        bugReporter = mock(BugReporter.class);
+        bugCaptor = ArgumentCaptor.forClass(BugInstance.class);
+    }
+
+    @AfterEach
+    void validate() {
+        Mockito.validateMockitoUsage();
+    }
+
+    @Test
+    void shouldMatchClassLevelSuppression() {
+        // given
+        matcher.addSuppressor(new ClassWarningSuppressor("UUF_UNUSED_FIELD", SuppressMatchType.DEFAULT, CLASS_ANNOTATION));
+        BugInstance bug = new BugInstance("UUF_UNUSED_FIELD", 1).addClass(CLASS_NAME);
+        // when
+        boolean matched = matcher.match(bug);
+        // then
+        assertThat("Should match the bug", matched, is(true));
```

Comment:
This ultimately doesn't give any more information than using `assertTrue`. If possible the error message should say more about the bug value.

Reply:
I'm not sure how to extract the reason _why_ the instance was matched or not, I think the test is fine like that